File: BRaVe-Mobile-Backend/Models/data_payload/Household.cs
using System.Text.Json.Serialization;

namespace BRaVe_Mobile_Backend.Models.data_payload
{
    public class Household
    {
        [JsonPropertyName("uuid")]
        public Guid uuid { get; set; }                     // UNIQUEIDENTIFIER

        [JsonPropertyName("householdId")]
        public string householdId { get; set; }               // VARCHAR(50)

        [JsonPropertyName("householdNo")]
        public int householdNo { get; set; }                   // INT NOT NULL

        [JsonPropertyName("individualNo")]
        public int individualNo { get; set; }                 // INT NOT NULL

        //public string activityCode;           // VARCHAR(10) NOT NULL

        [JsonPropertyName("registrationToken")]
        public string registrationToken { get; set; }          // NVARCHAR(255)

        [JsonPropertyName("householdSize")]
        public int householdSize { get; set; }                // INT NULL

        [JsonPropertyName("householdType")]
        public int householdType { get; set; }                // INT NULL

        [JsonPropertyName("address")]
        public object locationJson{ get; set; }               // NVARCHAR(250)

        [JsonPropertyName("fromServer")]
        public bool fromServer { get; set; } = false;

        [JsonPropertyName("isReadOnly")]
        public bool isReadOnly { get; set; } = false;

        public string collectionSource = "I";       // VARCHAR(1) NOT NULL (I/E)

        public string externalFamilyId { get; set; }           // NVARCHAR(100)

        [JsonPropertyName("dpAnswers")]
        public Dictionary<int, string> dataPoints { get; set; }                 // NVARCHAR(MAX)

        [JsonPropertyName("gps_coordinates")]
        public object gpsLatLonAccuracy { get; set; }          // NVARCHAR(150)

        public int tenantId { get; set; }                     // INT NOT NULL

        [JsonPropertyName("createdBy")]
        public string createdByUserId { get; set; }            // VARCHAR(50) NOT NULL

        [JsonPropertyName("createdOnMs")]
        public long createdOnMs { get; set; }                // DATETIME NOT NULL

        [JsonPropertyName("updatedBy")]
        public string updatedByUserId { get; set; }            // VARCHAR(50)

        [JsonPropertyName("updatedOnMs")]
        public long updatedOnMs { get; set; }               // DATETIME NULL

    }
}

File: BRaVeMobile/app/src/main/java/intl/iom/bravemobile/models/registrations/Household.java
package intl.iom.bravemobile.models.registrations;

import android.content.ContentValues;
import android.os.Build;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

import intl.iom.bravemobile.models.activities.ConsentsFeedback;

public class Household
{
    public String householdId;
    public int householdNo;
    public int individualNo;
    public UUID uuid = UUID.randomUUID();
    public String activityCode;
    public String registrationToken;
    public Integer householdSize;
    public Integer householdType;
    public String gps_coordinates;
    public LocationAddress address = new LocationAddress();
    public Map<Integer,String> dpAnswers;
    public ConsentsFeedback feedback;
    public String createdBy;
    public long createdOnMs;
    public String updatedBy;
    public long updatedOnMs;

    public db_staging_household getStagingData()
    {
        return new db_staging_household(householdId,householdNo,uuid,registrationToken,householdSize,householdType, gps_coordinates, address,dpAnswers, feedback, individualNo, createdBy, updatedBy, createdOnMs, updatedOnMs);
    }

    //non-db fields
    public List<Individual> individuals = new ArrayList<>();
    public boolean fromServer = false; //To know when downloaded from server
    public boolean isReadOnly = false; //If download from server and don't want alteration of the data
    public Date updated_on;
    //private Individual head = null;

    public boolean collect_head_only = false;


    public int individualCount()
    {
        if(individuals==null)
            return 0;

        return individuals.size();
    }

    public Individual getHead()
    {
        if(individuals.size()==0)
            return null;

        for (Individual i: individuals ) {

            if(i.relationship==0)
                return i;

        }

        return null;
    }

    public int getHouseholdSize()
    {

        if(householdSize==null)
            return 0;

        return householdSize;
    }


}
